SnowplowTracker/SnowplowTracker/Tracker.cs

```diff
@@ -0,0 +1,342 @@
+				}
+			} else if (eType == typeof(Unstructured)) {
+				Unstructured unstruct = (Unstructured)newEvent;
+				unstruct.SetBase64Encode(this.base64Encoded);
+				AddTrackerPayload ((TrackerPayload)unstruct.GetPayload (), contexts, eventId);
+			} else if (eType == typeof(Timing) || eType == typeof(ScreenView)) {
+				this.ProcessEvent(new Unstructured()
+				           .SetEventData((SelfDescribingJson)newEvent.GetPayload())
+				           .SetCustomContext(newEvent.GetContexts())
+				           .SetTimestamp(newEvent.GetTimestamp())
+				           .SetEventId(newEvent.GetEventId())
+				           .Build());
+			}
+		}
+
+		/// <summary>
+		/// Adds the tracker payload to the emitter.
+		/// </summary>
+		/// <param name="payload">The base event payload.</param>
+		/// <param name="contexts">The list of contexts from the event.</param>
+		private void AddTrackerPayload(TrackerPayload payload, List<IContext> contexts, string eventId) {
+
+			// Add default parameters to the payload
+			payload.Add (Constants.PLATFORM, this.platform.Value);
+			payload.Add (Constants.APP_ID, this.appId);
+			payload.Add (Constants.NAMESPACE, this.trackerNamespace);
+			payload.Add (Constants.TRACKER_VERSION, Version.VERSION);
+
+			// Add the subject data if available
+			if (subject != null) {
+				payload.AddDict (subject.GetPayload().GetDictionary());
+			}
+
+			// Add the session context if available
+			if (session != null) {
+				contexts.Add (session.GetSessionContext(eventId));
+			}
+
+			// Build the final context and add it to the payload
+			if (contexts != null && contexts.Count > 0) {
+				SelfDescribingJson envelope = GetFinalContext(contexts);
+				payload.AddJson (envelope.GetDictionary(), this.base64Encoded, Constants.CONTEXT_ENCODED, Constants.CONTEXT);
+			}
+
+			Log.Verbose ("Tracker: Sending event to the emitter.");
+			Log.Verbose (" + Event: " + payload.ToString());
+
+			// Add the event to the emitter.
+			emitter.Add (payload);
+		}
+
+		/// <summary>
+		/// Gets the final context.
+		/// </summary>
+		/// <returns>The final context.</returns>
+		/// <param name="contexts">A list of contexts to nest into the final context</param>
+		private SelfDescribingJson GetFinalContext(List<IContext> contexts) {
+			List<Dictionary<string, object>> contextDicts = new List<Dictionary<string, object>>();
+			foreach (IContext sdj in contexts) {
+				contextDicts.Add (sdj.GetJson().GetDictionary());
+			}
+			return new SelfDescribingJson (Constants.SCHEMA_CONTEXTS, contextDicts);
+		}
+
+		// --- Setters
+
+		/// <summary>
+		/// Sets the emitter. NOTE this is a blocking function as the emitter will not be replaced until it is no longer
+		/// sending events.
+		/// </summary>
+		/// <param name="emitter">Emitter.</param>
+		public void SetEmitter(Emitter emitter) {
+			// Prevents the emitter from looping
+			this.emitter.Stop ();
+
+			// Wait for the current loop to finish
+			while (emitter.GetSending()) {
+				Thread.Sleep (5);
+			}
```

Comment:
I think we could:

* Add a second table which is created fresh every time the Tracker is setup.
* This table will have the rowId of the event from the main table and a boolean field for each emitter attached.
* Once all emitters are true delete from events table and from this table.

In this way we don't need to persist an emitter as that could change overtime, emitters could be removed etc.

I do not think this would be too hard to implement for 0.1.0.